### PR TITLE
Dev/125 ExamScore Export report as CSV file

### PR DESCRIPTION
**Pull request by snowmancc**
#125 
image

revised
image

### CODE REVIEW

File: src/app/commons/buttons/ExportCsvButton.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import * as React from "react";
```
Comment (nyngwang) [inline src/app/commons/buttons/ExportCsvButton.js:1]: We can remove this line.

File: src/app/commons/buttons/ExportCsvButton.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import './ExportCsvButton.scss'
+
+//TODO
```
Comment (nyngwang) [inline src/app/commons/buttons/ExportCsvButton.js:4]: Should provide some description about what to do(in English). Notice that the format is like  
`// TODO: <your_description>`.

File: src/app/exam/score/ExamScorePage.js [hunk 3]
```diff
@@ -7,15 +7,26 @@ import {useExamContext} from "../questions/ExamContext";
...
+    const onButtonExportCSVClick = (examId) => {
+        examService.getExamTranscriptCsvFile(examId)
+        .then((res) => {
+            const url = window.URL.createObjectURL(new Blob([res.data]));
+            const link = document.createElement('a');
+            link.href = url;
+            link.setAttribute('download', 'ExamTranscript.csv');
+            link.click();
+        });
+    };
```
Comment (nyngwang) [inline src/app/exam/score/ExamScorePage.js:29]: We will move this function below the `useEffect(...)`(s). Function(s) not used by any `useEffect(...)` is prefered to be put after `useEffect(...)`(s).

**Review by nyngwang — CHANGES_REQUESTED** [review]
Review done!

**Review by Wally5077 — APPROVED** [review]
LGTM

File: src/app/commons/buttons/ExportCsvButton.js (hunk 2 above)
Comment (snowmancc) [reply src/app/commons/buttons/ExportCsvButton.js:4]: sorry, I forgot delete it.

File: src/app/commons/buttons/ExportCsvButton.scss [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  justify-content: center;
+  align-items: center;
+  padding: 10px;
+
+  position: static;
+  width: 190px;
+  height: 50px;
+  left: 0px;
+  top: 304px;
+
```
Comment (Johnny850807) [inline src/app/commons/buttons/ExportCsvButton.scss:18]: Do not use auto-layout !!!

use relative positioning

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

**Review by Wally5077 — APPROVED** [review]
LGTM

File: src/app/commons/buttons/ExportCsvButton.scss [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  display: flex;
+  border: none;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  padding: 10px;
+
+  position: relative;
+  width: 190px;
+  height: 50px;
```
Comment (Johnny850807) [inline src/app/commons/buttons/ExportCsvButton.scss:15]: no fixed points

File: src/app/commons/buttons/ExportCsvButton.scss [hunk 6]
```diff
@@ -0,0 +1,40 @@
+.export-blue-btn {
+  /* Export CSV */
+
+  /* Auto Layout */
```
Comment (Johnny850807) [inline src/app/commons/buttons/ExportCsvButton.scss:4]: delete

File: src/app/commons/buttons/ExportCsvButton.scss [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  position: relative;
+  width: 190px;
+  height: 50px;
+
+  background: #4585b4;
+  border-radius: 10px;
+
+  color: #ffffff;
+
+  /* Inside Auto Layout */
```
Comment (Johnny850807) [inline src/app/commons/buttons/ExportCsvButton.scss:22]: no

File: src/app/commons/buttons/ExportCsvButton.scss [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  flex-grow: 0;
+  margin: 12px 0px;
+
+  font-family: Poppins, sans-serif;
+  font-style: normal;
+  font-weight: 500;
+  font-size: 20px;
+  line-height: 30px;
+
+  /* identical to box height */
```
Comment (Johnny850807) [inline src/app/commons/buttons/ExportCsvButton.scss:36]: no

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

**Review by nyngwang — CHANGES_REQUESTED** [review]
Review donw.

**Review by nyngwang — CHANGES_REQUESTED** [review]
Review donw.

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Review by Chaoyuuu — CHANGES_REQUESTED** [review]
Review Done!

**Comment by Chaoyuuu** [comment]
First, I am really sorry about my late review and for asking to change lots of things. 😭 
There are my comments:
- The data in here is simple text, they are not `<FakeLink>`. Please change it.
image

- Please Remove the `<TableCell>`. We don't need it anymore. 
image

- It is not necessary to have `ExportCsvButton` component because we won't reuse it currently.
   
- Move these CSS styles in to a `.scss` file, including the CSS of `ExportCsvButton`.
image

File: src/app/exam/score/ExamScorePage.js [hunk 9]
```diff
@@ -33,30 +31,45 @@ const ExamScorePage = () => {
...
+
     return (
         <div className="exam-score">
-            <div className="font-poppins" style={{paddingTop: "20px", paddingBottom: "150px"}}>
-                <div className="mt-2" style={{display: "flex", justifyContent: "center"}}>
-                    <div className="mt-4 mr-3" style={{width: "fit-content", textAlign: "center"}}>
+            <div className="exam-score-font-poppins">
+                <div className="exam-score-mt-2">
+                    <div className="exam-score-mr-3">
```
Comment (Chaoyuuu) [inline src/app/exam/score/ExamScorePage.js:49]: `mt-2`, `mr-3`, `font-poppins` are the syntactic sugar provided by Bulma.
For example, `mt-2` means `margin-top: 0.5rem`.

That is, if the className is `exam-score-mt-2` then `mt-2` is out of action.

File: src/app/exam/score/ExamScorePage.js [hunk 10]
```diff
@@ -33,30 +31,45 @@ const ExamScorePage = () => {
...
+            link.href = url;
+            link.setAttribute('download', 'ExamTranscript.csv');
+            link.click();
+        });
+    };
+
     return (
         <div className="exam-score">
-            <div className="font-poppins" style={{paddingTop: "20px", paddingBottom: "150px"}}>
-                <div className="mt-2" style={{display: "flex", justifyContent: "center"}}>
```
Comment (Chaoyuuu) [inline src/app/exam/score/ExamScorePage.js:39]: Extract them to `<div className="exam-score">`.
These `<div>` tags are used for setting CSS, so we can extract them to one `<div>`.

**Review by Chaoyuuu — CHANGES_REQUESTED** [review]
Review Done!

**Review by Chaoyuuu — APPROVED** [review]
LGTM

**Review by Wally5077 — APPROVED** [review]
LGTM

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Pull request merged**